pkg/providers/instance/instance.go

```diff
@@ -84,8 +84,12 @@ func NewProvider(ctx context.Context, region string, ec2api ec2iface.EC2API, una
 }
 
 func (p *Provider) Create(ctx context.Context, nodeClass *v1beta1.EC2NodeClass, nodeClaim *corev1beta1.NodeClaim, instanceTypes []*cloudprovider.InstanceType) (*Instance, error) {
-	instanceTypes = p.filterInstanceTypes(nodeClaim, instanceTypes)
-	instanceTypes = orderInstanceTypesByPrice(instanceTypes, scheduling.NewNodeSelectorRequirements(nodeClaim.Spec.Requirements...))
+	schedulingRequirements := scheduling.NewNodeSelectorRequirements(nodeClaim.Spec.Requirements...)
+	// Only filter the instances if there are no minValues in the requirement.
+	if !schedulingRequirements.HasMinValues() {
+		instanceTypes = p.filterInstanceTypes(nodeClaim, instanceTypes)
+	}
+	instanceTypes = orderInstanceTypesByPrice(instanceTypes, schedulingRequirements)
 	if len(instanceTypes) > MaxInstanceTypes {
 		instanceTypes = instanceTypes[0:MaxInstanceTypes]
```

Comment:
Seems like we don't even need to truncate anymore if the upstream project is doing it for us

Reply:
You are right. Removed this.